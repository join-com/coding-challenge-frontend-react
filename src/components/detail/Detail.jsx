import React from 'react';

import dataItemType from '../types/dataItem';

import DataStateNotifier from '../../global/components/DataStateNotifier';

import DetailHeader from './DetailHeader';
import DetailInfos from './DetailInfos';

const Detail = ({
  item, ...otherProps
}) => (
  <DataStateNotifier
    {...otherProps}
  >
    <div className="Detail">
      <DetailHeader />
      <DetailInfos item={item} />
    </div>
  </DataStateNotifier>
);

export default Detail;

Detail.propTypes = {
  item: dataItemType,
};

Detail.defaultProps = {
  item: {},
};
