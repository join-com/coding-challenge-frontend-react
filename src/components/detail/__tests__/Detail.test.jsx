import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import Detail from '../Detail';

const { incidents } = staticData;

const props = {
  item: incidents[0],
};

describe('Detail', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<Detail {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should pass the error message to the DataStateNotifier component', () => {
    const tree = shallow(<Detail
      {...props}
      dataError="Oops something went wrong :("
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should pass the loading message to the DataStateNotifier component', () => {
    const tree = shallow(<Detail
      {...props}
      dataLoading="Loading data ..."
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
