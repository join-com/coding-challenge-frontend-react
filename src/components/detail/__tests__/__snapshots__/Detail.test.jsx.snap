// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should pass the error message to the DataStateNotifier component 1`] = `
<DataStateNotifier
  dataError="Oops something went wrong :("
  dataLoading=""
  resetDataError={[Function]}
>
  <div
    className="Detail"
  >
    <DetailHeader
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
    <DetailInfos
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
  </div>
</DataStateNotifier>
`;

exports[`Detail should pass the loading message to the DataStateNotifier component 1`] = `
<DataStateNotifier
  dataError=""
  dataLoading="Loading data ..."
  resetDataError={[Function]}
>
  <div
    className="Detail"
  >
    <DetailHeader
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
    <DetailInfos
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
  </div>
</DataStateNotifier>
`;

exports[`Detail should render the component correctly 1`] = `
<DataStateNotifier
  dataError=""
  dataLoading=""
  resetDataError={[Function]}
>
  <div
    className="Detail"
  >
    <DetailHeader
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
    <DetailInfos
      item={
        Object {
          "address": "Berlin, 12051, DE",
          "description": "Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ",
          "id": 94547,
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1543309200,
          "source": Object {
            "api_url": "https://bikeindex.org/api/v1/bikes/480323",
            "html_url": "https://bikeindex.org/bikes/480323",
            "name": "BikeIndex.org",
          },
          "title": "Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1550866132,
          "url": "https://bikewise.org/api/v1/incidents/94547",
        }
      }
    />
  </div>
</DataStateNotifier>
`;
